panther_battery/test/test_roboteq_republisher_node.cpp

```diff
@@ -0,0 +1,344 @@
+  driver_state_msg.rear.current = 0.1;
+
+  driver_state_pub_->publish(driver_state_msg);
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    roboteq_republisher_node_, battery_state_, std::chrono::milliseconds(1000)));
+
+  // check if message was sent correctly
+  auto expected_voltage = (driver_state_msg.front.voltage + driver_state_msg.rear.voltage) / 2.0;
+  auto expected_current = driver_state_msg.front.current + driver_state_msg.rear.current;
+  auto expected_percentage = (expected_voltage - 32.0) / (41.4 - 32.0);
+  CheckBatteryStateMsg(
+    expected_voltage, expected_current, expected_percentage,
+    BatteryStateMsg::POWER_SUPPLY_STATUS_DISCHARGING, BatteryStateMsg::POWER_SUPPLY_HEALTH_GOOD);
+
+  // sleep and check timeout
+  std::this_thread::sleep_for(std::chrono::milliseconds(2000));
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    roboteq_republisher_node_, battery_state_, std::chrono::milliseconds(1000)));
+
+  CheckBatteryStateMsg(
+    0.0, 0.0, 0.0, BatteryStateMsg::POWER_SUPPLY_STATUS_UNKNOWN,
+    BatteryStateMsg::POWER_SUPPLY_HEALTH_UNKNOWN);
+}
+
+TEST_F(TestRoboteqRepublisherNode, DriverStateMsgCANNetError)
+{
+  DriverStateMsg driver_state_msg;
+  driver_state_msg.front.voltage = 37.0;
+  driver_state_msg.rear.voltage = 37.0;
+  driver_state_msg.front.current = 0.1;
+  driver_state_msg.rear.current = 0.1;
+
+  driver_state_pub_->publish(driver_state_msg);
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    roboteq_republisher_node_, battery_state_, std::chrono::milliseconds(1000)));
+
+  // check if message was sent correctly
+  auto expected_voltage = (driver_state_msg.front.voltage + driver_state_msg.rear.voltage) / 2.0;
+  auto expected_current = driver_state_msg.front.current + driver_state_msg.rear.current;
+  auto expected_percentage = (expected_voltage - 32.0) / (41.4 - 32.0);
+  CheckBatteryStateMsg(
+    expected_voltage, expected_current, expected_percentage,
+    BatteryStateMsg::POWER_SUPPLY_STATUS_DISCHARGING, BatteryStateMsg::POWER_SUPPLY_HEALTH_GOOD);
+
+  // change values and set CAN network error true
+  driver_state_msg.front.fault_flag.can_net_err = true;
+  driver_state_msg.front.voltage = 39.0;
+  driver_state_msg.rear.voltage = 39.0;
+  driver_state_msg.front.current = 0.2;
+  driver_state_msg.rear.current = 0.2;
+  driver_state_pub_->publish(driver_state_msg);
+  ASSERT_TRUE(panther_utils::test_utils::WaitForMsg(
+    roboteq_republisher_node_, battery_state_, std::chrono::milliseconds(1000)));
+
+  // voltage and current values should not be updated
+  CheckBatteryStateMsg(
+    expected_voltage, expected_current, expected_percentage,
+    BatteryStateMsg::POWER_SUPPLY_STATUS_DISCHARGING, BatteryStateMsg::POWER_SUPPLY_HEALTH_GOOD);
+
+  // publish driver state until timeout is reached
+  rclcpp::Time start_time = roboteq_republisher_node_->now();
+  while (rclcpp::ok() &&
+         roboteq_republisher_node_->now() - start_time < std::chrono::milliseconds(2000)) {
```

Comment:
```suggestion
         roboteq_republisher_node_->now() - start_time < std::chrono::milliseconds(2500)) {
```
Can't is be a bit more? 2s will in worst case publish at most 1 time before reaching timeout